lighthouse-cli/printer.ts

```diff
@@ -17,45 +17,69 @@
 
 'use strict';
 
-const fs = require('fs');
-const ReportGenerator = require('../lighthouse-core/report/report-generator');
-const Formatter = require('../lighthouse-core/formatters/formatter');
-const log = require('../lighthouse-core/lib/log');
-
 /**
  * An enumeration of acceptable output modes:
- * <ul>
- *   <li>'pretty': Pretty print the results</li>
- *   <li>'json': JSON formatted results</li>
- *   <li>'html': An HTML report</li>
- * </ul>
- * @enum {string}
+ *   'pretty': Pretty print the results
+ *   'json': JSON formatted results
+ *   'html': An HTML report
  */
-const OUTPUT_MODE = {
-  pretty: 'pretty',
-  json: 'json',
-  html: 'html'
+enum OutputMode { pretty, json, html };
+type Mode = 'pretty' | 'json' | 'html';
+const Modes = ['pretty', 'json', 'html'];
```

Comment:
TypeScript enums can be used to map back to string values, as long as they aren't `const enum`s:

```ts
enum E { A, B, C }
console.log(E[E.A]); // 'A'
```

That's not always advisable – you get more code because you retain the strings, and this can lead to issues when using Closure Compiler in advanced mode. YMMV.